tasks/db_deploy/test/test_db_deploy.py

```diff
@@ -239,33 +253,585 @@ def return_cursor_return_function(conn: connection):
+            call(another_postgres_con_cursor, """SET SESSION AUTHORIZATION dbo;"""),
+            call(another_postgres_con_cursor, """SET SESSION AUTHORIZATION dbo;""")
         ], any_order=False)
+        self.assertEqual(5, mock_query_no_params.call_count)
+
+    @patch('boto3.client')
+    @patch('db_deploy.inner_task')
+    def test_task_pulls_correct_environment_variables(self,
+                                                      inner_task_mock,
+                                                      boto3_client_mock):
+        """
+        Verifies that environment variables are passed to inner_task correctly.
+        """
+        db_user_pass = uuid.uuid4().__str__()
+        db_admin_pass = uuid.uuid4().__str__()
+        db_host = uuid.uuid4().__str__()
+        db_name = uuid.uuid4().__str__()
+        db_user = uuid.uuid4().__str__()
+        db_port = uuid.uuid4().__str__()
+        drop_database = 'True'
+
+        ssm_vars = {
+            'drdb-user-pass': (db_user_pass, True),
+            'drdb-admin-pass': (db_admin_pass, True),
+            'drdb-host': (db_host, False)
+        }
+
+        ssm_mock = boto3_client_mock('ssm')
+
+        # noinspection PyPep8Naming
+        def ssm_return_function(Name, WithDecryption):
+            item = ssm_vars[Name]
+            if item[1] != WithDecryption:
+                raise KeyError
+            return {'Parameter': {'Value': item[0]}}
+
+        ssm_mock.get_parameter.side_effect = ssm_return_function
+
+        os.environ[db_deploy.OS_ENVIRON_DATABASE_NAME_KEY] = db_name
+        os.environ[db_deploy.OS_ENVIRON_DATABASE_USER_KEY] = db_user
+        os.environ[db_deploy.OS_ENVIRON_DATABASE_PORT_KEY] = db_port
+        os.environ[db_deploy.OS_ENVIRON_DROP_DATABASE_KEY] = drop_database
+
+        db_deploy.task()
+
+        inner_task_mock.assert_called_once_with(db_host, db_name, db_port, db_user, db_user_pass, db_admin_pass,
+                                                True)
+
+    @patch('db_deploy.get_db_connection')
+    @patch('db_deploy.create_database')
+    @patch('db_deploy.create_roles_and_users')
+    @patch('db_deploy.create_schema')
+    @patch('db_deploy.create_tables')
+    def test_inner_task_minimal_path(self,
+                                     mock_create_tables,
+                                     mock_create_schema,
+                                     mock_create_roles_and_users,
+                                     mock_create_database,
+                                     mock_get_db_connection):
+        """
+        A basic test of the function that orchestrates db deployment.
+        No additional need to create roles and users.
+        """
+        db_user_pass = uuid.uuid4().__str__()
+        db_admin_pass = uuid.uuid4().__str__()
+        db_host = uuid.uuid4().__str__()
+        db_name = uuid.uuid4().__str__()
+        db_user = uuid.uuid4().__str__()
+        db_port = uuid.uuid4().__str__()
```

Comment:
May want to use closer to valid data for port. port should be a number. For PostgreSQL default is 5432 but could use a random int converted to string.

Reply:
Makes sense. Fixed.